src/main/java/updater/UpdateManager.java

```diff
@@ -90,6 +96,30 @@ private boolean initUpdate() {
             return false;
         }
 
+        return extractReplaceAndExecuteJar();
+    }
+
+    private boolean extractReplaceAndExecuteJar() {
+        logger.info("Extracting restarter JAR");
+        File replaceAndExecuteJar = new File(UPDATE_DIRECTORY + File.separator + UPDATE_RESTARTER_APP_NAME);
+
+        if (!replaceAndExecuteJar.exists()) {
+            try {
+                replaceAndExecuteJar.createNewFile();
+            } catch (IOException e) {
+                logger.error("Can't create replacer-and-executor JAR");
+                return false;
+            }
+        }
+
+        try (InputStream in = UpdateManager.class.getClassLoader().getResourceAsStream("updater/replaceAndExecute");
+             OutputStream out = new FileOutputStream(replaceAndExecuteJar)) {
+            IOUtils.copy(in, out);
+        } catch (IOException e) {
+            logger.error("Can't copy replacer-and-executor JAR", e);
+            return false;
+        }
```

Comment:
I didn't make any changes to this because this method does one thing, which is copying a file from the resource to a local file. I take that part of copying file to a target file includes creating such target file if it does not exist yet.

I kept the boolean returns because this is an initialisation sequence. I can make it throw exception and handle the exception in the caller, but it seems better to know if extracting the resource to a local file was successful or not rather than throw exception.